.github/workflows/deploy.yml

```diff
@@ -0,0 +1,188 @@
+name: Deploy Images
+
+on:
+  push:
+    paths:
+      - 'pkg/version/version.go'
+    branches:
+      - main
+      - 'release/**'
+
+  workflow_dispatch:
+
+env:
+  DOCKERHUB_IMAGE: solarwinds/solarwinds-otel-collector
+
+jobs:
+  get_current_tag:
+    runs-on: ubuntu-latest
+    outputs:
+      image_tag: ${{ steps.get-tag.outputs.value }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Get Current Tag
+        id: get-tag
+        run: echo "tag=v$(grep -oP '(?<=const Version = ")[^"]+' "../../pkg/version/version.go")" >> $GITHUB_OUTPUT
+
+  build_and_test:
+    uses: ./.github/workflows/buildAndTest.yml
```

Comment:
The `Deploy Images` workflow (executed when merging new image version to `main`/`release`) depends on the `Build and Test Images` workflow. The latter should be executed in whole, including Windows images.